internal/handler/Handler.go

```diff
@@ -0,0 +1,323 @@
+		return
+
+	case r.Method == http.MethodDelete && positionRe.MatchString(r.URL.Path):
+		h.deletePosition(w, r)
+		return
+	case r.Method == http.MethodDelete && employeeRe.MatchString(r.URL.Path):
+		h.deleteEmployee(w, r)
+		return
+
+	case r.Method == http.MethodPut && createPositionRe.MatchString(r.URL.Path):
+		h.updatePosition(w, r)
+		return
+	case r.Method == http.MethodPut && createEmployeeRe.MatchString(r.URL.Path):
+		h.updateEmployee(w, r)
+		return
+	default:
+		fmt.Println("error: not found")
+		http.NotFound(w, r)
+		return
+	}
+}
+
+func (h *Handler) getPositions(w http.ResponseWriter, r *http.Request) {
+	query, err := url.ParseQuery(r.URL.RawQuery)
+	limit, err := strconv.ParseInt(query["limit"][0], 10, 64)
+	if err != nil {
+		http.Error(w, "error with server", http.StatusInternalServerError)
+		return
+	}
+	offset, err := strconv.ParseInt(query["offset"][0], 10, 64)
+	if err != nil {
+		http.Error(w, "error with server", http.StatusInternalServerError)
+		return
+	}
+	positions, err := h.service.GetPositions(int(limit), int(offset))
+	if err != nil {
+		if err.Error() == "incorrect data"{
+			http.Error(w, "error: not found", http.StatusNotFound)
+			return
+		}else {
+			http.Error(w, "error with server", http.StatusInternalServerError)
+			return
+		}
+	}
+	jsonBytes, err := json.Marshal(positions)
+	if err != nil {
+		http.Error(w, "error with server", http.StatusInternalServerError)
+		return
+	}
+	_, er := w.Write(jsonBytes)
+	if er != nil {
+		http.Error(w, "error with server", http.StatusInternalServerError)
+	}
+	w.WriteHeader(http.StatusOK)
+}
+
+func (h *Handler) getEmployees(w http.ResponseWriter, r *http.Request) {
+	query, err := url.ParseQuery(r.URL.RawQuery)
+	limit, err := strconv.ParseInt(query["limit"][0], 10, 64)
+	if err != nil {
+		http.Error(w, "error with server", http.StatusInternalServerError)
+		return
+	}
+	offset, err := strconv.ParseInt(query["offset"][0], 10, 64)
+	if err != nil {
+		http.Error(w, "error with server", http.StatusInternalServerError)
+		return
+	}
+	employees, err := h.service.GetEmployees(int(limit), int(offset))
+	if err != nil {
+		if err.Error() == "incorrect data"{
+			http.Error(w, "error: not found", http.StatusNotFound)
+			return
+		}else {
+			http.Error(w, "error with server", http.StatusInternalServerError)
+			return
+		}
+	}
+	jsonBytes, err := json.Marshal(employees)
+	if err != nil {
+		http.Error(w, "error with server", http.StatusInternalServerError)
+		return
+	}
+	_, er := w.Write(jsonBytes)
+	if er != nil {
+		http.Error(w, "error with server", http.StatusInternalServerError)
+	}
+	w.WriteHeader(http.StatusOK)
+}
+
+func (h *Handler) getPosition(w http.ResponseWriter, r *http.Request) {
+	matches := positionRe.FindStringSubmatch(r.URL.Path)
+	if len(matches) < 2 {
+		http.NotFound(w, r)
+		return
+	}
+	p, err := h.service.GetPosition(matches[1])
+	if err != nil {
+		w.WriteHeader(http.StatusNotFound)
+		_, err := w.Write([]byte("Position not found"))
```

Comment:
` h.service.GetPosition()` should return a special error in case it couldn't find position. 
Then you should check which an error you get from `h.service.GetPosition()` and only in case if there is "position not found" you should return `position not found` with 404 code.

The code you've written returns StatusNotFound in all cases.

Reply:
+